mm2src/coins/lp_coins.rs

```diff
@@ -236,6 +236,36 @@ pub enum NegotiateSwapContractAddrErr {
     NoOtherAddrAndNoFallback,
 }
 
+#[derive(Serialize, Display, SerializeErrorType)]
+#[serde(tag = "error_type", content = "error_data")]
+pub enum GetRawTransactionError {
+    #[display(fmt = "Internal error `{}`", _0)]
+    Internal(String),
+    #[display(fmt = "Invalid response `{}`", _0)]
+    InvalidResponse(String),
+    #[display(fmt = "No activated coin with such name `{}`", _0)]
+    InvalidCoin(String),
+    #[display(fmt = "No valid transaction with such hash `{}`", _0)]
+    InvalidTx(String),
+    #[display(fmt = "Invalid hash `{}`", _0)]
+    InvalidTxHash(String),
+    #[display(fmt = "No 'coin' field")]
+    NoCoinField,
+    #[display(fmt = "No 'tx_hash' field")]
+    NoTxHashField,
+    Transport(String),
+}
+
+impl HttpStatusCode for GetRawTransactionError {
+    fn status_code(&self) -> StatusCode {
+        match self {
+            GetRawTransactionError::Internal(_) => StatusCode::INTERNAL_SERVER_ERROR,
+            GetRawTransactionError::InvalidResponse(_) => StatusCode::BAD_GATEWAY,
+            _ => StatusCode::BAD_REQUEST,
```

Comment:
We prefer to use explicit pattern matching to have a compilation error when we add new enum variants.